clouddq/classes/dq_configs_cache.py

```diff
@@ -486,17 +473,80 @@ def _resolve_bigquery_entity_uri(
-        )
+            return False
+        else:
+            dataplex_entity = dataplex_entities_match[0]
+            clouddq_entity = dq_entity.DqEntity.from_dataplex_entity(
+                entity_id=entity_uri.get_db_primary_key(),
+                dataplex_entity=dataplex_entity,
+            )
         return clouddq_entity
+
+    def _resolve_bigquery_entity_uri(
+        self,
+        entity_uri: dq_entity_uri.EntityUri,
+        client: clouddq_dataplex.CloudDqDataplexClient,
+        bigquery_client: BigQueryClient,
+        enable_experimental_bigquery_entity_uris: bool = True,
+    ) -> dq_entity.DqEntity:
+        if not enable_experimental_bigquery_entity_uris:
+            raise NotImplementedError(
+                f"entity_uri '{entity_uri.complete_uri_string}' "
+                "has unsupported scheme 'bigquery://'.\n"
+                "Use CLI flag --enable_experimental_bigquery_entity_uris "
+                "to enable looking up bigquery:// entity_uri scheme in format "
+                "bigquery://projects/<project-id>/datasets/<dataset-id>/tables/<table-id> "
+                "schemes using Dataplex Metadata API.\n"
+                "Ensure the BigQuery dataset containing this table "
+                "is registered as an asset in Dataplex.\n"
+                "You can then specify the corresponding Dataplex "
+                "projects/locations/lakes/zones as part of the "
+                "metadata_default_registries YAML configs, e.g.\n"
+                f"{SAMPLE_DEFAULT_REGISTRIES_YAML}"
+            )
+        required_arguments = ["projects", "datasets", "tables"]
+        for argument in required_arguments:
+            uri_argument = entity_uri.get_configs(argument)
+            if not uri_argument:
+                raise RuntimeError(
+                    f"Failed to retrieve default Bigquery '{argument}' for "
+                    f"entity_uri: {entity_uri.complete_uri_string}. \n"
+                    f"'{argument}' is a required argument to look-up metadata for the entity_uri "
+                    "using Bigquery API.\n"
+                )
+
+        project_id = entity_uri.get_configs("projects")
+        table_name = entity_uri.get_table_name()
+        bq_table_exists = bigquery_client.is_table_exists(
+            table=table_name, project_id=project_id
+        )
+        if bq_table_exists:
+            logger.debug(
+                f"The Table '{table_name}' in the "
+                f"specified entity_uri '{entity_uri}' "
+                f"exists in Bigquery."
+            )
+            dataplex_entity = self.is_dataplex_entity(
+                entity_uri=entity_uri, client=client
+            )
+            if dataplex_entity:
+                clouddq_entity = dataplex_entity
+            else:
+                clouddq_entity = dq_entity.DqEntity.from_bq_entity_uri(
+                    entity_uri=entity_uri,
+                    bigquery_client=bigquery_client,
+                )
```

Comment:
Can we please add an integration test for this behaviour?

Reply:
added the test for this behaviour